app/models/journeyDomain/item/supplyChainActors/SupplyChainActorDomain.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.journeyDomain.item.supplyChainActors
+
+import cats.implicits.catsSyntaxTuple2Semigroupal
+import models.journeyDomain.Stage.{AccessingJourney, CompletingJourney}
+import models.journeyDomain.{GettableAsReaderOps, JourneyDomainModel, Stage, UserAnswersReader}
+import models.{Index, Mode, SupplyChainActorType, UserAnswers}
+import pages.item.supplyChainActors.index.{IdentificationNumberPage, SupplyChainActorTypePage}
+import play.api.i18n.Messages
+import play.api.mvc.Call
+import uk.gov.hmrc.http.HttpVerbs.GET
+
+case class SupplyChainActorDomain(role: SupplyChainActorType, identification: String)(itemIndex: Index, actorIndex: Index) extends JourneyDomainModel {
+
+  def asString(implicit messages: Messages): String =
+    SupplyChainActorDomain.asString(role, identification)
```

Comment:
Any reason this can't be `override def toString: String = s"$role - $identification"` ?

Reply:
What you could do is add
```
def asString(implicit messages: Messages): String = messages(s"$messageKeyPrefix.$this")
```
to `Radioable`

Reply:
Then your asString method in here would be `def asString(implicit messages: Messages): String = s"${role.asString} - $identification"